### PR TITLE
Check counterparty port id during the channel handshake

### LINKED ISSUE #281: Check counterparty port id during the channel handshake
Check counterparty port id during the channel handshake, i.e., https://github.com/cosmos/interchain-security/blob/eccbfaae9cdc39a769d94814eece02a2eafe333f/x/ccv/parent/module.go#L236 and https://github.com/cosmos/interchain-security/blob/eccbfaae9cdc39a769d94814eece02a2eafe333f/x/ccv/child/module.go#L221

See spec, i.e., https://github.com/cosmos/ibc/blob/main/spec/app/ics-028-cross-chain-validation/methods.md#ccv-ccf-coinit1 and https://github.com/cosmos/ibc/blob/main/spec/app/ics-028-cross-chain-validation/methods.md#ccv-pcf-cotry1

### PR DESCRIPTION
**Pull request by smarshall-spitzbart**
Closes #281 and extends #287, by adding e2e tests for the provider module's ibc handshake methods, and improving coverage of of ```ConsumerTestSuite.TestOnChanOpenAck```

@mpoke I tried to get all your changes in from the last PR but the merge conflicts were tricky, lmk if I missed any of your ideas

### CODE REVIEW

File: x/ccv/consumer/ibc_module.go [hunk 1]
```diff
@@ -94,10 +103,10 @@ func (am AppModule) OnChanOpenAck(
 	ctx sdk.Context,
 	portID,
 	channelID string,
-	counterpartyChannelID string,
+	_ string, // Counter party channel ID is unused per spec
```
Comment (smarshall-spitzbart) [inline x/ccv/consumer/ibc_module.go:106]: Curious why the IBCModule interface enforces us to pass in a counter party channel ID here? [Spec](https://github.com/cosmos/ibc/blob/main/spec/app/ics-028-cross-chain-validation/methods.md#ccv-ccf-coack1) doesn't mention it. Is there uses for it in other IBC modules?

Comment (mpoke) [reply x/ccv/consumer/ibc_module.go:106]: @AdityaSripal should we check here the `counterpartyChannelID`? Or is it already done by core IBC?

File: e2e-tests/valset_update_test.go [hunk 2]
```diff
@@ -8,9 +8,8 @@ import (
 	clienttypes "github.com/cosmos/ibc-go/v3/modules/core/02-client/types"
 	channeltypes "github.com/cosmos/ibc-go/v3/modules/core/04-channel/types"
 	appConsumer "github.com/cosmos/interchain-security/app/consumer"
-	consumertypes "github.com/cosmos/interchain-security/x/ccv/consumer/types"
-	providertypes "github.com/cosmos/interchain-security/x/ccv/provider/types"
 	"github.com/cosmos/interchain-security/x/ccv/types"
+	ccv "github.com/cosmos/interchain-security/x/ccv/types"
```
Comment (mpoke) [inline e2e-tests/valset_update_test.go:12]: Duplicate lines.

File: e2e-tests/setup_test.go [hunk 3]
```diff
@@ -18,8 +18,7 @@ import (
 	appConsumer "github.com/cosmos/interchain-security/app/consumer"
 	appProvider "github.com/cosmos/interchain-security/app/provider"
 	"github.com/cosmos/interchain-security/testutil/simapp"
-	consumertypes "github.com/cosmos/interchain-security/x/ccv/consumer/types"
-	providertypes "github.com/cosmos/interchain-security/x/ccv/provider/types"
+
 	"github.com/cosmos/interchain-security/x/ccv/types"
```
Comment (mpoke) [inline e2e-tests/setup_test.go:22]: Same as line 10.

**Review by mpoke — APPROVED** [review]
LGTM. Just some nits bellow.

**Review by jtremback — APPROVED** [review]
LGTM

**Pull request merged**